test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,141 @@
+// +build e2e
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	apimachinerytypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/test/helpers/kind"
+	"sigs.k8s.io/cluster-api/test/helpers/scheme"
+	"sigs.k8s.io/cluster-api/util"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestE2E(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "e2e Suite")
+}
+
+const (
+	capiNamespace      = "capi-system"
+	capiDeploymentName = "capi-controller-manager"
+	setupTimeout       = 10 * 60
+)
+
+var (
+	managerImage    = flag.String("managerImage", "", "Docker image to load into the kind cluster for testing")
+	capiComponents  = flag.String("capiComponents", "", "URL to CAPI components to load")
+	kustomizeBinary = flag.String("kustomizeBinary", "kustomize", "path to the kustomize binary")
+
+	kindCluster kind.Cluster
+	kindClient  crclient.Client
+	suiteTmpDir string
+)
+
+var _ = ginkgo.BeforeSuite(func() {
```

Comment:
Is there a way to opt out of the before and after suites?

Reply:
Not that I'm aware of, but I wouldn't expect other providers to inherit this suite directly, but instead we'd continue to move common logic into the helpers, and have each provider implement their own suites.